src/formatDate.js

```diff
@@ -1,14 +1,54 @@
 'use strict';
 
-/**
- * @param {string} date
- * @param {string[]} fromFormat
- * @param {string[]} toFormat
- *
- * @returns {string}
- */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const prevSplitter = fromFormat[3];
+  const newSplitter = toFormat[3];
+  const rawDate = date.split(prevSplitter);
+  const newRawDate = [];
+
+  function setTheDate(dateType, nextDateType = dateType) {
+    newRawDate[toFormat.indexOf(nextDateType)] =
+      rawDate[fromFormat.indexOf(dateType)];
+  }
+
+  setTheDate('DD');
+  setTheDate('MM');
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YYYY')) {
+    setTheDate('YYYY');
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    setTheDate('YYYY', 'YY');
+
+    newRawDate[toFormat.indexOf('YY')] = newRawDate[
+      toFormat.indexOf('YY')
+    ].slice(2, 4);
+  }
```

Comment:
When converting from 'YYYY' to 'YY', the code correctly slices the last two digits. Ensure that this logic is applied consistently across all relevant parts of the code.